tests/runner.py

```diff
@@ -124,13 +124,22 @@ def get_default_pattern(loader):
     # Get all test names (optionally from some path with some pattern)
     with PerfContext() as pc:
         # the files are searched from `tests/` folder, starting with `test_`
-        files = glob.glob(os.path.join(os.path.dirname(__file__), "test_*.py"))
+        tests_path = Path(__file__).parent / args.path
+        files = {
+            file.relative_to(tests_path).as_posix()
+            for file in tests_path.rglob("test_*py")
+            if re.search(args.pattern, file.name)
```

Comment:
"dist" still wasn't researched correctly here because of the file suffix.

Reply:
```suggestion
            if re.search(args.pattern, file.name[:-3])
```
I think this will fix it here.